lib/ansible/galaxy/collection/__init__.py

```diff
@@ -144,12 +120,27 @@ class PkgReq:  # type: ignore[no-redef]
 SIGNATURE_COUNT_RE = r"^(?P<strict>\+)?(?:(?P<count>\d+)|(?P<all>all))$"
 
 
+class DisplayThread:
+
+    def __init__(self, display_queue: DisplayQueueType) -> None:
+        self.display_queue = display_queue
+
+    def __getattr__(self, attr: str) -> t.Callable:
+        def call_display(*args, **kwargs) -> None:
+            self.display_queue.put((attr, args, kwargs))
+
+        return call_display
+
+
+display: Display | DisplayThreadProto = Display()
+
+
 @dataclass
 class ManifestControl:
-    directives: list[str] = None
+    directives: list[str]
```

Comment:
This seems like a user-facing change, shouldn't it remain optional?

```suggestion
    directives: list[str] | None = None
```

Reply:
This class is only used to pass things between calls internally, though. But I get your point. MyPy wanted me to do something where it's initialized, I think. I'll double-check if I can revert this bit.